test/PaginatorTest.php

```diff
@@ -919,6 +919,23 @@ public function testPaginatorShouldProduceDifferentCacheIdsWithDifferentAdapterI
         $this->assertNotEquals($firstCacheId, $secondCacheId);
     }
 
+    public function testPaginatorShouldProduceDifferentCacheIdsDependingOnGetArrayCopy(): void
+    {
+        $paginator                    = new Paginator\Paginator(new TestAsset\TestAdapter('foo'));
+        $reflectionGetCacheInternalId = new ReflectionMethod($paginator, '_getCacheInternalId');
+        $reflectionGetCacheInternalId->setAccessible(true);
+        /** @var string $firstCacheId */
+        $firstCacheId = $reflectionGetCacheInternalId->invoke($paginator);
```

Comment:
Not happy with how this is tested - we should try reaching this path from a public API

Reply:
All the tests of _getCacheInternalId are coded like this. I just did the same way.

Reply:
Ah, ugly, but then I think it's OK :+1: 

Please don't write tests like these in the wild, but we should roll with your fix as-is then